src/test/suite/extension.test.ts

```diff
@@ -6,18 +6,84 @@
 import {messages} from '../../lib/messages';
 
 // You can import and use all API from the 'vscode' module
 // as well as import your extension to test it
 import * as vscode from 'vscode';
-// import * as myExtension from '../../extension';
 
 suite('Extension Test Suite', () => {
 	vscode.window.showInformationMessage('Start all tests.');
+	// Note: __dirname is used here because it's consistent across file systems.
+	const codeFixturesPath: string = path.resolve(__dirname, '..', '..', '..', 'code-fixtures');
+
+	suite('E2E', () => {
+		let ext = vscode.extensions.getExtension('salesforce.sfdx-code-analyzer-vscode');
+		let context: vscode.ExtensionContext;
+		suiteSetup(async () => {
+			// Activate the extension.
+			context = await ext.activate();
+		});
+
+		setup(() => {
+			// Verify that there are no existing diagnostics floating around.
+			const diagnosticsArrays = vscode.languages.getDiagnostics();
+			for (const [uri, diagnostics] of diagnosticsArrays) {
+				expect(diagnostics, `${uri.toString()} should start without diagnostics`).to.be.empty;
+			}
+		});
+
+		teardown(async () => {
+			// Close any open tabs and close the active editor.
+			await vscode.commands.executeCommand('workbench.action.closeAllEditors');
+			// Also, clear any diagnostics we added.
+			_clearDiagnostics();
+		});
+
+		// The use of `this.timeout` requires us to use `function() {}` syntax instead of arrow functions.
+		// (Arrow functions bind lexical `this` and we don't want that.)
+		test('sfca.runOnActiveFile', async function() {
+			// ===== SETUP =====
+			// Set the timeout to a frankly absurd value, just to make sure Github Actions
+			// can finish it in time.
+			this.timeout(60000);
+			// Open a file in the editor.
+			const fileUri: vscode.Uri = vscode.Uri.file(path.join(codeFixturesPath, 'folder-a', 'MyClassA1.cls'));
+			const doc = await vscode.workspace.openTextDocument(fileUri);
+			await vscode.window.showTextDocument(doc);
+
+			// ===== TEST =====
+			// Run the "scan active file" command.
+			await vscode.commands.executeCommand('sfca.runOnActiveFile');
+
+			// ===== ASSERTIONS =====
+			// Verify that we added diagnostics.
+			const diagnosticArrays = vscode.languages.getDiagnostics();
+			const [uri, diagnostics] = diagnosticArrays.find(uriDiagPair => uriDiagPair[0].toString() === fileUri.toString());
+			expect(uri, `Expected diagnostics for ${fileUri.toString()}`).to.exist;
+			expect(diagnostics, 'Expected non-empty diagnostic array').to.not.be.empty;
+
+			// At present, we expect only violations for PMD's `ApexDoc` rule.
+			for (const diagnostic of diagnostics) {
+				expect(diagnostic.source).to.equal('pmd via Code Analyzer', 'Wrong source');
+				expect(diagnostic.code).to.have.property('value', 'ApexDoc', 'Wrong rule violated');
+			}
+		});
+
+		test('sfca.runOnSelected', async function() {
+			// TODO: Add actual tests for `runOnSelected`.
```

Comment:
If we still want to do the other story for adding further integration/E2E tests, they can be added in these TODOs.